fava/json_api.py

```diff
@@ -95,28 +95,79 @@ def add_document():
     return _api_success(message='Uploaded to {}'.format(filepath))
 
 
-@json_api.route('/add-transaction/', methods=['PUT'])
-def add_transaction():
-    """Add a transaction."""
-    json = request.get_json()
-
+def _add_transaction(json):
     postings = []
+
     for posting in json['postings']:
         if posting['account'] not in g.ledger.attributes.accounts:
             return _api_error('Unknown account: {}.'
                               .format(posting['account']))
         number = D(posting['number']) if posting['number'] else None
         amount = Amount(number, posting.get('currency'))
-        postings.append(Posting(posting['account'], amount,
-                                None, None, None, None))
+        postings.append(data.Posting(posting['account'], amount,
+                        None, None, None, None))
 
     if not postings:
         return _api_error('Transaction contains no postings.')
 
     date = util.date.parse_date(json['date'])[0]
-    transaction = Transaction(
+    transaction = data.Transaction(
         json['metadata'], date, json['flag'], json['payee'],
         json['narration'], None, None, postings)
 
-    g.ledger.file.insert_transaction(transaction)
+    g.ledger.file.insert_entry(transaction)
+
+
+def _add_balance(json):
```

Comment:
> Maybe tackle this in a separate PR?

👍  the functionality needed for this PR could simplify reviewing it (and with tests, it could be merged even if there's no frontend using it yet)